### PR TITLE
Refactor umaps module

### PR DESCRIPTION
**Pull request by roshankern**
This PR is ready for review!

In this PR, umpas are made for each feature type: CP, DP, CP_and_DP.
Also, instead of saving the pngs of the umaps, the umaps are simply created and stored in the jupyter notebook. In the future, the umap coordinate data will be saved in tidy long format (outside the scope of this PR).

### CODE REVIEW

File: 4.analyze_data/README.md [hunk 1]
```diff
@@ -6,20 +6,18 @@ In this module, we analyze the normalized training data features from [3.normali
 
 We use [UMAP](https://github.com/lmcinnes/umap) for analyis of features.
 UMAP was introduced in [McInnes, L, Healy, J, 2018](https://arxiv.org/abs/1802.03426) as a manifold learning technique for dimension reduction.
-We use UMAP to reduce the feature data from 1280 features to 1, 2, and 3 dimensions.
-We use [Matplotlib](https://matplotlib.org/) to visualize the 1D, 2D, and 3D UMAPS.
+We use UMAP to reduce the feature data to 1 and 2 dimensions.
```
Comment (axiomcura) [inline 4.analyze_data/README.md]:
```suggestion
We use UMAP to reduce the feature data into 1 and 2 dimensions.
```

File: 4.analyze_data/scripts/nbconverted/analyze_data.py [hunk 2]
```diff
@@ -96,53 +78,27 @@
 class_colors_1 = get_class_colors(classes_1, "rainbow")
 class_colors_2 = get_class_colors(classes_2, "bright")
```
Comment (axiomcura) [inline 4.analyze_data/scripts/nbconverted/analyze_data.py:82]:
```python 

classes_2 = [
    # "Large",
    "Prometaphase",
    "Grape",
    "Interphase",
    "Apoptosis",
    "ADCCM",
    # "Folded",
    # "SmallIrregular",
    # "Polylobed",
    # "Metaphase",
    # "Binuclear",
    # "Hole",
    # "Anaphase",
    # "MetaphaseAlignment",
    # "Elongated",
    # "OutOfFocus",
]


```
It seems that some classes are commented out in the `classes_2` variable. Will these features be used in the future?

File: 4.analyze_data/scripts/nbconverted/analyze_data.py [hunk 3]
```diff
@@ -96,53 +78,27 @@
...
 
-umap_3D_save_path = pathlib.Path(f"{save_dir}/norm_train_3D_umap_other.png")
-show_3D_umap(feature_data, phenotypic_classes, class_colors_2, save_path=umap_3D_save_path, point_size=point_size, alpha=alpha)
+feature_types = ["CP", "DP", "CP_and_DP"]
+
+for feature_type in feature_types:
+    print(f"Showing UMAPs created with {feature_type} features")
+    
+    metadata_dataframe, feature_data = split_data(training_data, feature_type)
+    phenotypic_classes = metadata_dataframe["Mitocheck_Phenotypic_Class"]
```
Comment (axiomcura) [inline 4.analyze_data/scripts/nbconverted/analyze_data.py:97]: This is more of a discussion point. Would it be best to add a small comment to explain how the data is being split here? If not, feel free to ignore this?

**Review by axiomcura — APPROVED** [review]
Nice :D! Added some comments. great job!

File: 4.analyze_data/scripts/nbconverted/analyze_data.py (hunk 2 above)
Comment (roshankern) [reply 4.analyze_data/scripts/nbconverted/analyze_data.py:82]: These classes could be used in the future, depending on what classes we want to be colored in the final umaps. I'll added a comment explaining how to use this variable in  https://github.com/WayScience/mitocheck_data/pull/30/commits/753097d11e6034e6432da9517c7d759ddf79f43c

File: 4.analyze_data/scripts/nbconverted/analyze_data.py (hunk 3 above)
Comment (roshankern) [reply 4.analyze_data/scripts/nbconverted/analyze_data.py:97]: I think you are right! Added in https://github.com/WayScience/mitocheck_data/pull/30/commits/df8d13427419072a6547f29e73fb75f66ecf2ca0

**Pull request merged**